File: androidApp/src/main/kotlin/dev/code93/emvqr/domain/generator/TlvEncoder.kt
package dev.code93.emvqr.domain.generator

/**
 * Codificador TLV del estándar EMVCo: tag de 2 caracteres + longitud de
 * 2 dígitos + valor. Los templates anidados usan el mismo formato.
 */
object TlvEncoder {
    fun tlv(tag: String, value: String): String {
        require(tag.length == 2) { "El tag debe tener 2 caracteres: $tag" }
        require(value.length in 1..99) { "El valor del tag $tag debe tener entre 1 y 99 caracteres" }
        return tag + value.length.toString().padStart(2, '0') + value
    }
}

File: iosApp/iosApp/Domain/Generator/TlvEncoder.swift
import Foundation

/// Codificador TLV del estándar EMVCo: tag de 2 caracteres + longitud de
/// 2 dígitos + valor. Los templates anidados usan el mismo formato.
enum TlvEncoder {
    static func tlv(_ tag: String, _ value: String) -> String {
        precondition(tag.count == 2, "El tag debe tener 2 caracteres: \(tag)")
        precondition((1...99).contains(value.count), "El valor del tag \(tag) debe tener entre 1 y 99 caracteres")
        return tag + String(format: "%02d", value.count) + value
    }
}
